frontend/javascripts/oxalis/view/left-border-tabs/modals/add_volume_layer_modal.tsx

```diff
@@ -17,11 +17,67 @@ import {
 import {
   getAllReadableLayerNames,
   getVolumeTracingLayers,
-  validateReadableLayerName,
 } from "oxalis/model/accessors/volumetracing_accessor";
+import messages from "messages";
 import InputComponent from "oxalis/view/components/input_component";
 import api from "oxalis/api/internal_api";
 import Toast from "libs/toast";
+
+export type ValidationResult = { isValid: boolean; message: string };
+export function checkForLayerNameDuplication(
+  readableLayerName: string,
+  allReadableLayerNames: string[],
+): ValidationResult {
+  const layerNameDoesNotExist = _.countBy(allReadableLayerNames)[readableLayerName] <= 0;
+  return {
+    isValid: layerNameDoesNotExist,
+    message: layerNameDoesNotExist ? "" : messages["tracing.volume_layer_name_duplication"],
+  };
+}
+
+export function checkLayerNameForInvalidCharacters(readableLayerName: string): ValidationResult {
+  const uriSaveCharactersRegex = /[0-9a-zA-Z-._~]+/g;
+  // Removing all URISaveCharacters from readableLayerName. The left over chars are all invalid.
+  const allInvalidChars = readableLayerName.replace(uriSaveCharactersRegex, "");
+  const allUniqueInvalidCharsAsSet = new Set(...allInvalidChars);
```

Comment:
```suggestion
  const allUniqueInvalidCharsAsSet = new Set(allInvalidChars);
```

otherwise only the first character will end up in the Set, compare the [MDN page](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set/Set).

Reply:
Oh, you are right... I thought I did a quick tryout of this in the browser console. 🙈